src/test/java/org/opensrp/web/rest/SettingResourceTest.java

```diff
@@ -116,81 +137,47 @@ public void setUp() {
-		SettingSearchBean sQB = new SettingSearchBean();
-		sQB.setTeamId("my-team-id");
-		sQB.setTeam(null);
-		sQB.setLocationId(null);
-		sQB.setProviderId(null);
-		sQB.setServerVersion(1000L);
-		
-		settingService.findSettings(sQB);
-		Mockito.verify(settingRepository, Mockito.times(1)).findSettings(sQB);
-		Mockito.verifyNoMoreInteractions(settingRepository);
-		
-	}
-	
-	@Test
-	public void testSaveSetting() throws Exception {
-		String documentId = "1";
-		Mockito.doNothing().when(settingRepository).add(Matchers.any(SettingConfiguration.class));
-		settingService.saveSetting(settingJson);
-		
-		Mockito.verify(settingRepository, Mockito.times(1)).add(settingConfigurationArgumentCaptor.capture());
-		Mockito.verify(settingRepository, Mockito.times(1)).get(documentId);
-		Mockito.verifyNoMoreInteractions(settingRepository);
-	}
-	
-	@Test
-	public void testUpdateSetting() throws Exception {
-		String documentId = "settings-document-id-2";
-		Mockito.when(settingRepository.get("settings-document-id-2")).thenReturn(new SettingConfiguration());
-		Mockito.doNothing().when(settingRepository).update(Matchers.any(SettingConfiguration.class));
-		
-		settingService.saveSetting(settingJsonUpdate);
-		
-		Mockito.verify(settingRepository, Mockito.times(1)).get(documentId);
-		Mockito.verify(settingRepository, Mockito.times(1)).update(settingConfigurationArgumentCaptor.capture());
-		Mockito.verifyNoMoreInteractions(settingRepository);
-	}
-	
-	@Test
-	public void testAddServerVersion() throws Exception {
-		
-		settingService.addServerVersion();
-		Mockito.verify(settingRepository, Mockito.times(1)).findByEmptyServerVersion();
-		Mockito.verifyNoMoreInteractions(settingRepository);
+				.param(Event.TEAM_ID, "my-team-id").param(Event.PROVIDER_ID, "demo")).andExpect(status().isOk()).andReturn();
+
+		String responseString = result.getResponse().getContentAsString();
+		if (responseString.isEmpty()) {
+			fail("Test case failed");
+		}
+		JsonNode actualObj = mapper.readTree(responseString);
+		assertEquals(actualObj.get(0).get("identifier").asText(), EXPECTED_IDENTFIER);
+		assertEquals(actualObj.get(0).get("teamId").asText(), EXPECTED_TEAM_ID);
+		assertEquals(actualObj.size(), 1);
 	}
-	
+
 	@Test
 	public void testValidValue() throws Exception {
 		SettingConfiguration settingConfiguration = getSettingConfigurationObject();
 		assertNotNull(settingConfiguration);
 		assertEquals("site_characteristics", settingConfiguration.getIdentifier());
 		assertEquals("my-team-id", settingConfiguration.getTeamId());
 	}
-	
+
+	@Test
+	public void testPostSaveSetting() throws Exception {
+		String SETTINGS_JSON = "{\"settingConfigurations\":[]}";
+
+		when(settingService.saveSetting(Matchers.any(String.class))).thenReturn("ID-12345");
+		MvcResult mvcResult = this.mockMvc.perform(
+				post(BASE_URL + "/sync").contentType(MediaType.APPLICATION_JSON).body(SETTINGS_JSON.getBytes()).accept(MediaType.APPLICATION_JSON))
+				.andExpect(status().isOk()).andReturn();
```

Comment:
assert the service class was called with the correct params

Reply:
Done.